src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -391,12 +391,17 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
                     }
                     else
                     {
-                        // Send an empty result so that processing does not hang
+                        // Send an empty result so that processing does not hang when peek def service called from non-mssql clients
```

Comment:
Technically I think the non-mssql client comment belongs on the else condition you added below. Since `ShouldSkipIntellisense` will return true for non-mssql editors